internal/provisioner/kops_provisioner.go

```diff
@@ -31,6 +31,7 @@ type ProvisioningParams struct {
 	DeployMinioOperator     bool
 	NdotsValue              string
 	PGBouncerConfig         *PGBouncerConfig
+	EtcdManagerEnv          map[string]interface{}
```

Comment:
nit: I'm not a fan of `interfaces{}` where we know the value types (even if the flag is `int`, internally we only use those as strings). 

And I understand that this is a map instead of a struct to ease the iteration later on, but still seems weird to me. In this case I would go with a `EtcdManager` struct with the fields we allow via flags, and an helper method to convert the structure to a manageable `[]string` to use with `kops.SetCluster`. This is personal preference!

Reply:
Right. As values are always `string` in **Env**, we can use `map[string]string` here. But I'm not sure about `struct`. If we use struct, we need pointer types, or else we can't make any flag optional.

```go
type EtcdManager struct {
	etcdQuotaBackendBytes *int
	etcdListenMetricsURL *string
}
```

Reply:
I'm like 0/5 on the struct, just felt more familiar. Leaving this up to you, since you've dug more in that particular piece on the provisioner.